packages/force_update/lib/src/force_update_guard.dart

```diff
@@ -56,51 +70,57 @@ class _ForceUpdateGuardState extends State<ForceUpdateGuard> {
     init();
   }
 
-  Future<void> init() async {
-    _packageInfo = await PackageInfo.fromPlatform();
-
-    unawaited(_updateVersionsInfo());
-
-    _checkForEnforcedUpdateTimer = Timer.periodic(
-      ForceUpdateGuard.updateCheckingInterval,
-      (_) => _updateVersionsInfo(),
-    );
-
-    final mostRecentForceUpdateResult = await _storage.readMostRecentResult();
+  Future<void> _applyResult({
+    required ForceUpdateResult? result,
+  }) async {
     final currentVersion = Version.parse(_packageInfo.version);
 
-    if (mostRecentForceUpdateResult == null ||
-        mostRecentForceUpdateResult.versionAtTimeOfRequest < currentVersion) {
+    if (result == null || result.versionAtTimeOfRequest < currentVersion) {
       return;
     }
 
-    force.value = mostRecentForceUpdateResult.result ==
-        VersionSupportResultDTO.updateRequired;
+    force.value = result.conclusion == VersionSupportResultDTO.updateRequired;
 
-    if (mostRecentForceUpdateResult.result ==
-        VersionSupportResultDTO.updateSuggested) {
+    if (result.conclusion == VersionSupportResultDTO.updateSuggested) {
       if (widget._actuallyUseAndroidSystemUI) {
         await InAppUpdate.checkForUpdate();
         await InAppUpdate.startFlexibleUpdate();
         return InAppUpdate.completeFlexibleUpdate();
+      } else {
+        widget.controller._suggest.value = true;
       }
+    }
+  }
 
-      final context = widget.dialogContextKey.currentContext;
+  Future<void> _updateAndMaybeApplyVersionsInfo() async {
+    await _updateVersionsInfo();
+
+    final recentResult = await _storage.readMostRecentResult();
+    return switch ((
```

Comment:
IMO normal if would be more readable here

Reply:
I thought the same until I tried it - the names are very long and I'm not sure if it actually helps
```
    if ((recentResult?.conclusion == VersionSupportResultDTO.updateRequired &&
            widget.showForceUpdateScreenImmediately) ||
        (recentResult?.conclusion == VersionSupportResultDTO.updateSuggested &&
            widget.showSuggestUpdateDialogImmediately)) {
      return _applyResult(result: recentResult);
    }
```